language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java

```diff
@@ -225,13 +225,13 @@ public LedgerConfigurationClient getLedgerConfigurationClient() {
     }
 
     @Override
-    public TimeClient getTimeClient() { return timeClient; }
+    public TimeClient getTimeClient() {
+        return timeClient;
+    }
 
     public void close() throws Exception {
         channel.shutdownNow();
-        while (!channel.awaitTermination(1, TimeUnit.SECONDS)) {
-
-        }
+        channel.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
```

Comment:
This seems like a long time... :)

Reply:
It's way shorter then what it replaces. :wink:

Reply:
True, however wouldn't the original code throw after a second?

Reply:
Nah, `awaitTermination` just returns `false`, which then loops around for another attempt.